core/src/main/java/io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java

```diff
@@ -754,6 +785,16 @@ void processMeasurementEvent(WriterJob job) {
                             break;
                         }
 
+                        case ColumnType.GEOSHORT: // these are storage tags produced by the reader threads
+                        case ColumnType.GEOINT:   // rather than emit a specific NewLineProtoParser.ENTITY_TYPE_GEOHASH
+                        case ColumnType.GEOLONG:  // and then switch over the storageTag, we simply emit the later.
+                        case ColumnType.GEOBYTE: {
+                            long geohash = Unsafe.getUnsafe().getLong(bufPos);
+                            bufPos += Long.BYTES;
+                            row.putGeoHash(colIndex, entityType, geohash);
```

Comment:
This is back to point Vlad made, here should be not one method in writer but 4 different for every GEO switch

```
row.putGeoHashByte();
row.putGeoHashShort() 
row.putGeoHashInt() 
row.putGeoHashLong()
```

Or simply

```
...
case NewLineProtoParser.GEOSHORT:
   row.putShort(colIndex, geohash);
   break;

case NewLineProtoParser.GEOINT:
   row.putInt(colIndex, geohash);
   break;

...
```

Also I'd not use ColumnType here but encode them in 4 unrelated NewLineProtoParser constants in the reader threads. It's longer code but it's better seen how things are related by constant names

Reply:
yes, thank you for this, I did not think of that for some reason :P